services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -421,6 +421,35 @@ async function maybeQueueQueryCmrWorkItem(
   }
 }
 
+
+/**
+ * Check to see if a workflow step for the given job is completed
+ *
+ * @param logger - the logger to use for logging timing statements
+ * @param tx - the database transaction to use for making queries
+ * @param workItem - the work-item o
+ */
+async function isStepComplete(logger: Logger, tx: Transaction, workItem: WorkItem, thisStep: WorkflowStep): Promise<boolean> {
+  let notComplete = await (await logAsyncExecutionTime(
```

Comment:
It's probably worth mentioning why we are doing this (either as a comment or in the docstring) - it's because this first check is much faster, but there are cases where it won't find any incomplete work items, but we're not done with the step. So we do the fast check, then if it thinks we might be done, we do the guaranteed check.

Reply:
Added additional clarifying comment